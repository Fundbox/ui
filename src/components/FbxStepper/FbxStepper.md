# FbxStepper
A component for representing numbered steps with text labels

`steps: String[]` - Should be an array of step strings.

`activeStep: Number` - Should be the index of one of the steps.
