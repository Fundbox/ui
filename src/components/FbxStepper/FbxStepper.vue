<template>
  <ul class="steps-container layout-row">
    <li v-for="(step, index) in steps" :key="index + step" class="step flex" :class="{ active: isStepActive(index), done: isStepDone(index) }">
      <div class="step-number-wrapper"><span class="step-number">{{ index + 1 }}</span></div>
      <div class="step-label">{{ step }}</div>
    </li>
  </ul>
</template>

<script>
export default {
  name: 'FbxStepper',
  props: {
    steps: {
      type: Array,
      required: true,
    },
    activeStep: {
      type: Number,
      default: 0,
    }
  },
  methods: {
    isStepActive(index) {
      return index === this.activeStep
    },
    isStepDone(index) {
      return index < this.activeStep
    },
  },
}
</script>

<style lang="scss" scoped>
@import "~@fundbox/ui/es/styles/utils/utilities.scss";

$stepNumberSize: 24px;

.step {
  text-align: center;
  color: $medium-blue;
  font-size: 13px;

  &:not(:first-child) .step-number-wrapper:before, &:not(:last-child) .step-number-wrapper:after {
    content: "";
  }

  &.active  {
    .step-number {
      background: $medium-blue;
      color: $white;
    }
  }

  &.done .step-number{
    background: $medium-blue;
    &:before {
      content: url("~@fundbox/ui/es/assets/v-icon.min.svg");
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -45%);
    }
  }
}

.step-number {
  position: relative;
  display: inline-block;
  border: 2px solid $medium-blue;
  background: $white;
  border-radius: 50%;
  width: $stepNumberSize;
  height: $stepNumberSize;
  line-height: $stepNumberSize - 2px;
}

.step-number-wrapper {
  position: relative;

  &:before, &:after {
    position: absolute;
    display: inline-block;
    top: 50%;
    transform: translateY(-50%);
    height: 2px;
    width: 50%;
    background: $medium-blue;
  }

  &:before {
    left: 0;
  }

  &:before {
    right: 0;
  }
}

.step-label {
  padding-top: 10px;
  font-weight: 300;
}
</style>
