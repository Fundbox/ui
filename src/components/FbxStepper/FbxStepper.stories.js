import { storiesOf } from '@storybook/vue'
import { withInfo } from 'storybook-addon-vue-info'

import FbxStepper from './FbxStepper.vue'
import summary from './FbxStepper.md'

const stories = storiesOf('Components/Stepper', module)

stories.add('default', withInfo({ summary })(() => ({
  components: { FbxStepper },
  data() {
    return {
      steps: ['step one', 'step two', 'step three'],
    }
  },
  template: `<fbx-stepper :steps="steps" :active-step="1"></fbx-stepper>`
})))
