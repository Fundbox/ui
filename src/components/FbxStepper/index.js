import FbxStepper from './FbxStepper.vue'
import { registerComponents, vueUse } from '../../utils/plugins'

const components = {
  FbxStepper
}

const VuePlugin = {
  install(Vue) {
    registerComponents(Vue, components)
  }
}

vueUse(VuePlugin)

export default VuePlugin
export { FbxStepper }
