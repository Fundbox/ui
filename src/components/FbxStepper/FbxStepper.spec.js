import { shallowMount } from '@vue/test-utils'
import FbxStepper from './FbxStepper.vue'

const steps = ['step one','step two','step three']

describe('Components/FbxStepper', () => {
  describe('snapshots', () => {
    it('renders the default correctly', () => {
      const wrapper = shallowMount(FbxStepper, {
        propsData: {
          steps,
        }
      })
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('renders active state correctly', () => {
      const wrapper = shallowMount(FbxStepper, {
        propsData: {
          steps,
          activeStep: 1,
        }
      })
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
