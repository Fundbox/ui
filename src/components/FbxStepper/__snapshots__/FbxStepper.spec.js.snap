// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/FbxStepper snapshots renders active state correctly 1`] = `
<ul class="steps-container layout-row">
  <li class="step flex done">
    <div class="step-number-wrapper"><span class="step-number">1</span></div>
    <div class="step-label">step one</div>
  </li>
  <li class="step flex active">
    <div class="step-number-wrapper"><span class="step-number">2</span></div>
    <div class="step-label">step two</div>
  </li>
  <li class="step flex">
    <div class="step-number-wrapper"><span class="step-number">3</span></div>
    <div class="step-label">step three</div>
  </li>
</ul>
`;

exports[`Components/FbxStepper snapshots renders the default correctly 1`] = `
<ul class="steps-container layout-row">
  <li class="step flex active">
    <div class="step-number-wrapper"><span class="step-number">1</span></div>
    <div class="step-label">step one</div>
  </li>
  <li class="step flex">
    <div class="step-number-wrapper"><span class="step-number">2</span></div>
    <div class="step-label">step two</div>
  </li>
  <li class="step flex">
    <div class="step-number-wrapper"><span class="step-number">3</span></div>
    <div class="step-label">step three</div>
  </li>
</ul>
`;
